### PR TITLE
fix: support UUIDs as per RFC spec (#50)

### PR DESCRIPTION
**Pull request by aikrom**
## Proposed changes

Added support uppercase/lowercase UUIDs as per RFC spec.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/adonisjs/http-server/blob/master/.github/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/Router/Matchers.ts [hunk 1]
```diff
@@ -30,8 +30,11 @@ export class RouteMatchers extends Macroable implements RouteMatchersContract {
   /**
    * Enforce value to be formatted as uuid
    */
-  public uuid(): { match: RegExp } {
-    return { match: /^[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}$/ }
+  public uuid(): { match: RegExp; cast: (value: string) => string } {
+    return {
+      match: /^[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}$/,
+      cast: (value) => value.toLowerCase(),
```
Comment (Barbapapazes) [inline src/Router/Matchers.ts:36]: Are-you sure? You meas toUpperCase or you must update the regex to use lower case instead of upper case.

Comment (Barbapapazes) [reply src/Router/Matchers.ts:36]: or you can use a regex to match both : 

```js
const regex = /^[0-9a-zA-F]{8}-[0-9a-zA-F]{4}-[0-9a-zA-F]{4}-[0-9a-zA-F]{4}-[0-9a-zA-F]{12}$/
```

Comment (thetutlage) [reply src/Router/Matchers.ts:36]: Yup. I think you are still only allowing uppercase values. So regex must be updated

Comment (aikrom) [reply src/Router/Matchers.ts:36]: Somehow missed it. Today I will commit the corrected regex.

Comment (aikrom) [reply src/Router/Matchers.ts:36]: I think now you can do the review again

**Comment by thetutlage** [comment]
Thanks! Finally, let's have a test that validates case insensitive uuid and casts it to lower

**Comment by thetutlage** [comment]
Merging, as I am already working on a release. I will add the required tests myself

**Pull request merged**